### ISSUE TITLE
Feature Request - Ability to ignore warnings per file

### ISSUE THREAD

**Issue by ChickenSwimmer2020** [markdownlint/markdownlint#540]
## Describe the Enhancement:
I want to be able to ignore specific files for certain issues, like ignoring MD032 on Markdown.md, while allowing MD032 on Markdown2.md

### Impacted Rules:
None I think

## Describe the Need:
I think people who like different markdown files to follow different syntaxing rules would like this, being able to disable warnings for only certain files individually

## Current Alternative
Not that I'm aware of

## Can We Help You Implement This?:
Sadly, no. I hold no understanding of how Visual Studio Code plugins work.

**Comment by nbehrnd** [comment]
An example in Linux Debian 14/forky which provides the syntax checker packaged as `ruby-mdl` / `markdownlint` ([tracker page]) with the executable `mdl`:

```bash
$ ls
mystyle.rb  README1.md  README2.md  README3.md  README.md
$ mdl README[12].md -s mystyle.rb 
README1.md:1: MD022 Headers should be surrounded by blank lines
README1.md:4: MD022 Headers should be surrounded by blank lines
README2.md:22: MD012 Multiple consecutive blank lines
README2.md:5: MD022 Headers should be surrounded by blank lines
README2.md:11: MD031 Fenced code blocks should be surrounded by blank lines
```

This approach partially relies some regex like(?) expansion by the shell instead of a globbing `mdl *.md`.  The second part is the definition of a style file

```bash
$ cat mystyle.rb 
all
exclude_rule 'MD032'
```

which can be used at local level, per directory -- for instance for a project --, or across the operating system.  The repository documents the rules and syntax in `docs/creating_styles.md` and provides an example of application (see `example/exclude_example`).  Though tailored to Debian, its [man page] equally may be helpful, too.

To ease a replication, the .zip archive below includes the test data and the style file applied.

[tracker page]: https://tracker.debian.org/pkg/ruby-mdl
[man page]: https://manpages.debian.org/testing/markdownlint/markdownlint.1.en.html

[test_case.zip](https://github.com/user-attachments/files/26361431/test_case.zip)

**Comment by ChickenSwimmer2020** [comment]
As helpful as that looks/may be, I dont use Debian and have no clue how to apply all that on windows

**Comment by jaymzh** [comment]
@nbehrnd - the OP is asking to be able to do that per-file, which isn't something we support. What you lay out is a per-repo configuration.

@ChickenSwimmer2020 - it's not linux specific. It's just putting a config file in your repo using whatever text editor you like on whatever platform you're on.

**Comment by ChickenSwimmer2020** [comment]
Right, and I have said config json already, I just think that there should be config options to restrict per-file

**Comment by jaymzh** [comment]
@ChickenSwimmer2020 if you have a JSON config, then you're not using this markdownlint, you're using https://github.com/DavidAnson/markdownlint

Wrong project. And it does have per-file options.

**Comment by ChickenSwimmer2020** [comment]
oh my god your right ive been using that plugin, im so sorry. thanks for pointing me in the right direction
